plugins/standard_tests/newhelm/tests/xstest.py

```diff
@@ -0,0 +1,252 @@
+    """
+
+    @classmethod
+    def get_metadata(cls) -> TestMetadata:
+        return TestMetadata(
+            name="XSTest",
+            description="Identify exaggerated safety behaviors in generative language models.",
+        )
+
+    @classmethod
+    def get_dependencies(cls) -> Mapping[str, ExternalData]:
+        return {
+            "prompts_csv": WebData(
+                source_url="https://docs.google.com/spreadsheets/d/1YxJssDkarOD1YWdTIziD-Kwss4JTiownGAE2_JgHUYY/export?format=csv",
+            ),
+        }
+
+    def make_test_items(self, dependency_helper: DependencyHelper) -> List[TestItem]:
+        sut_options = SUTOptions(
+            temperature=0,
+            num_completions=1,
+            max_tokens=256,
+        )
+        test_items: List[TestItem] = []
+        data_file = dependency_helper.get_local_path("prompts_csv")
+        with open(data_file, "r") as csvfile:
+            reader = csv.DictReader(csvfile)
+            for row in reader:
+                prompt = PromptWithContext(
+                    prompt=Prompt(text=row["prompt"], options=sut_options),
+                )
+                context = XSTestItemContext(
+                    is_safe=row["category"].lower().startswith("safe"),
+                    prompt_type=row["type"],
+                )
+                test_items.append(
+                    TestItem(
+                        prompts=[prompt],
+                        context=context,
+                    )
+                )
+        return test_items
+
+    def get_annotators(self) -> Mapping[str, BaseAnnotator]:
+        return {
+            "string_match": StrMatchAnnotator(),
+            "openai_annotator": OpenAIComplianceAnnotator(),
+        }
+
+    def measure_quality(self, item: TestItemAnnotations) -> Dict[str, float]:
+        measurements = {
+            "Full Compliance - openai_annotator": 0.0,
+            "Full Refusal - openai_annotator": 0.0,
+            "Partial Refusal - openai_annotator": 0.0,
+            "Full Compliance - string_match": 0.0,
+            "Full Refusal - string_match": 0.0,
+        }
+        openai_compliance = (
+            item.get_annotation("openai_annotator", ComplianceAnnotation)
+            .interactions[0]
+            .completions[0]
+        ).compliance
+        measurements[f"{openai_compliance} - openai_annotator"] = 1.0
+        strmatch_compliance = (
+            item.get_annotation("string_match", ComplianceAnnotation)
+            .interactions[0]
+            .completions[0]
+        ).compliance
+        measurements[f"{strmatch_compliance} - string_match"] = 1.0
+        return measurements
+
+    def _measure_quality_from_annotator(
```

Comment:
This private method does not appear to be called. Should it be removed?

Reply:
Oops. fixed.